Graphics/GraphicsEngine/interface/PipelineState.h

```diff
@@ -340,6 +344,7 @@ struct GraphicsPipelineDesc
               SubpassIndex      == Rhs.SubpassIndex      &&
               ShadingRateFlags  == Rhs.ShadingRateFlags  &&
               DSVFormat         == Rhs.DSVFormat         &&
+              UseReadOnlyDSV    == Rhs.UseReadOnlyDSV    &&
```

Comment:
Still, I don't like the idea of additional fields. Can we find a value for the render pass based on the DepthStencilState?
@TheMostDiligent

Reply:
We _can_ do that, but it will (1) silently break existing user code, and (2) will require extra render target changes in user code. Which is the trade off I am personally not willing to take.

Reply:
> Still, I don't like the idea of additional fields.

Neither do I.

>  Can we find a value for a render pass based on DepthStencilState?

Probably. We can check the `DepthWriteEnable` flag and create the read-only depth. However this will create a problem with binding normal DSV and using it with the read-only depth pipeline.
I was thinking to have two views inside each DSV - one normal, one readonly, but then again there is a problem with mixing normal and read-only pipelines.

In current approach, what happens if you bind read-only depth pipeline, set render targets and then use read-write depth pipeline?

We have the `SetRenderTargetsExt` call. It looks like we will need to add ReadOnlyDepth flag to the struct it takes.

Reply:
@TheMostDiligent 
Sorry, I don’t follow your question. The act of binding pipeline and setting render targets are still independent from each other. It’s okay if you have mismatched pipeline bound at some point as long as you don’t use it.

This is the reason why I don’t want to make automatic flagless detection — it’s gonna mess this up. We will have to have two pipelines in pipeline object if depth write is disabled. One if user decides to use PSO with read write depth and another if with read only. And this will also create non trivial engagement between setting PSO and render target. I would even say that effective PSO will have to be resolved in prepareDraw.